diff --git a/app/models/concerns/attachmentable.rb b/app/models/concerns/attachmentable.rb
index a83e178fc..ba5144f7e 100644
--- a/app/models/concerns/attachmentable.rb
+++ b/app/models/concerns/attachmentable.rb
@@ -5,7 +5,7 @@ require 'mime/types/columnar'
 module Attachmentable
   extend ActiveSupport::Concern
 
-  MAX_MATRIX_LIMIT = 33_177_600 # 7680x4320px or approx. 847MB in RAM
+  MAX_MATRIX_LIMIT = 37_748_736 # 6144x6144px
   GIF_MATRIX_LIMIT = 921_600    # 1280x720px
 
   # For some file extensions, there exist different content
diff --git a/app/models/media_attachment.rb b/app/models/media_attachment.rb
index 6708cd779..dcf71bc6a 100644
--- a/app/models/media_attachment.rb
+++ b/app/models/media_attachment.rb
@@ -70,7 +70,7 @@ class MediaAttachment < ApplicationRecord
 
   IMAGE_STYLES = {
     original: {
-      pixels: 8_294_400, # 3840x2160px
+      pixels: 37_748_736, # 6144x6144px
       file_geometry_parser: FastGeometryParser,
     }.freeze,
 
