diff --git a/app/javascript/mastodon/features/emoji/emoji.js b/app/javascript/mastodon/features/emoji/emoji.js
index c11ef458c..e762ba8eb 100644
--- a/app/javascript/mastodon/features/emoji/emoji.js
+++ b/app/javascript/mastodon/features/emoji/emoji.js
@@ -74,7 +74,7 @@ const emojifyTextNode = (node, customEmojis) => {
       const filename = autoPlayGif ? custom_emoji.url : custom_emoji.static_url;
       replacement = document.createElement('img');
       replacement.setAttribute('draggable', 'false');
-      replacement.setAttribute('class', 'emojione custom-emoji');
+      replacement.setAttribute('class', 'emojione custom-emoji custom-emoji-scaleable');
       replacement.setAttribute('alt', shortcode);
       replacement.setAttribute('title', shortcode);
       replacement.setAttribute('src', filename);
