diff --git a/app/models/custom_emoji.rb b/app/models/custom_emoji.rb
index 717de2772..4ba8f496d 100644
--- a/app/models/custom_emoji.rb
+++ b/app/models/custom_emoji.rb
@@ -24,7 +24,7 @@
 class CustomEmoji < ApplicationRecord
   include Attachmentable
 
-  LIMIT = 256.kilobytes
+  LIMIT = 4.megabytes
 
   SHORTCODE_RE_FRAGMENT = '[a-zA-Z0-9_]{2,}'
 
@@ -33,7 +33,7 @@ class CustomEmoji < ApplicationRecord
     (?=[^[:alnum:]:]|$)/x
   SHORTCODE_ONLY_RE = /\A#{SHORTCODE_RE_FRAGMENT}\z/
 
-  IMAGE_MIME_TYPES = %w(image/png image/gif image/webp).freeze
+  IMAGE_MIME_TYPES = %w(image/png image/gif image/webp image/jpeg image/svg+xml).freeze
 
   belongs_to :category, class_name: 'CustomEmojiCategory', optional: true
 
diff --git a/config/locales/en.yml b/config/locales/en.yml
index 71b8f27aa..97c4fa375 100644
--- a/config/locales/en.yml
+++ b/config/locales/en.yml
@@ -327,7 +327,7 @@ en:
       enable: Enable
       enabled: Enabled
       enabled_msg: Successfully enabled that emoji
-      image_hint: PNG or GIF up to %{size}
+      image_hint: PNG, GIF, WebP, SVG or JPEG up to %{size}
       list: List
       listed: Listed
       new:
