diff --git a/app/javascript/mastodon/features/compose/components/poll_form.jsx b/app/javascript/mastodon/features/compose/components/poll_form.jsx
index a2036d0cc5..cca3b60760 100644
--- a/app/javascript/mastodon/features/compose/components/poll_form.jsx
+++ b/app/javascript/mastodon/features/compose/components/poll_form.jsx
@@ -82,7 +82,7 @@ const Option = ({ multipleChoice, index, title, autoFocus }) => {
 
       <AutosuggestInput
         placeholder={intl.formatMessage(messages.option_placeholder, { number: index + 1 })}
-        maxLength={50}
+        maxLength={10000}
         value={title}
         lang={lang}
         spellCheck
diff --git a/app/validators/poll_options_validator.rb b/app/validators/poll_options_validator.rb
index fd29fc1b44..a871b69a53 100644
--- a/app/validators/poll_options_validator.rb
+++ b/app/validators/poll_options_validator.rb
@@ -2,7 +2,7 @@
 
 class PollOptionsValidator < ActiveModel::Validator
   MAX_OPTIONS      = 4
-  MAX_OPTION_CHARS = 50
+  MAX_OPTION_CHARS = 10000
 
   def validate(poll)
     poll.errors.add(:options, I18n.t('polls.errors.too_few_options')) unless poll.options.size > 1
